modules/Microsoft.ContainerInstance/containerGroups/deploy.bicep

```
@@ -4,15 +4,24 @@ param name string
 @description('Required. The containers and their respective config within the container group.')
 param containers array
 
-@description('Optional. Ports to open on the public IP address. Must include all ports assigned on container level.')
+@description('Required. Ports to open on the public IP address. Must include all ports assigned on container level.')
```

If this is required, it should not have a default value.

Is it?

---

Ports are required if `ipAddressType` is set to `public`.
So lets change this to Conditional?
```suggestion
@description('Conditional. Ports to open on the public IP address. Must include all ports assigned on container level. Required if `ipAddressType` is set to `public`.')
```

---

Conditional it is, yes.